linopy/expressions.py

```diff
@@ -266,17 +284,16 @@ def __radd__(self, other):
         if other == 0:
             return self
         else:
-            return NotImplemented
+            return self.__add__(other)
```

Comment:
Do not do this!

`__add__` is preferred over `__radd__` anyway by the operations engine of python, unless `self` is of a type that inherits from `other`. In the latter case after `NotImplemented` python tries `__add__` anyway. 

Check https://docs.python.org/3.6/library/constants.html#NotImplemented and https://docs.python.org/3.6/library/numbers.html#implementing-the-arithmetic-operations for details.

Reply:
Ok good to know, sounds logical, but when trying it out the interpreter complains about not knowing what to do with "float + LinearExpr"... Can you get the left addition of constants to work like that?

Reply:
Ok I see calling __add__ is rather horrible, but radd needs to cover the case where other is a constant... So probably be clean about it and write the radd logic out?

Reply:
Sorry, i was just going over a lot of different work in parallel, so i did not grasp the deeper meaning of what you were trying to do! Should have asked!

It makes sense now to me.

In your particular case i would go for
```
__radd__ = __add__
```
to clarify that the same code should be used for right addition than for general.

Otherwise i would have tried to abstract. That particular way of course does not work for the rsub, so i guess after all the way you are calling between the methods is actually fine.

And the confusion is mine. Apologies.